### PR TITLE
Fix wrong smallest key of delete range tombstones

### PR DESCRIPTION
**Pull request by huachaohuang**
Since tombstones are not stored in order, we may get a wrong smallest key if we only consider the first added tombstone.
Check https://github.com/facebook/rocksdb/issues/2752 for more details.

### CODE REVIEW

File: db/range_del_aggregator.cc [hunk 1]
```diff
@@ -412,7 +412,6 @@ void RangeDelAggregator::AddToBuilder(
 
   // Note the order in which tombstones are stored is insignificant since we
   // insert them into a std::map on the read path.
-  bool first_added = false;
```
Comment (ajkr) [inline db/range_del_aggregator.cc]: could you have just moved this one line lower? Within each stripe the tombstones should be sorted. So we only need to consider the first tombstone in each stripe.

File: db/range_del_aggregator_test.cc [hunk 2]
```diff
@@ -149,6 +149,38 @@ TEST_F(RangeDelAggregatorTest, AlternateMultipleAboveBelow) {
        {"h", 0}});
 }
 
+TEST_F(RangeDelAggregatorTest, UnorderedTombstones) {
```
Comment (ajkr) [inline db/range_del_aggregator_test.cc]: Also this belongs in db_range_del_test

**Comment by ajkr** [comment]
Excellent, thanks everyone at PingCAP who helped find this bug :).

**Comment by zhangjinpeng1987** [comment]
@ajkr Can you please release a version after this pr has been merged, combine with this [bug fix](https://github.com/facebook/rocksdb/pull/2745)?

**Comment by ajkr** [comment]
@zhangjinpeng1987 - sure, the 5.7.3 release is available now containing both fixes.

**Pull request closed without merging**